src/IotHub/IotHub/IotHub/DataPlane/DistributedTracing/SetAzIotHubDistributedTracing.cs

```diff
@@ -0,0 +1,129 @@
+        [ValidateRange(0,100)]
+        [Alias("Rate")]
+        public int SamplingRate { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (ShouldProcess(this.DeviceId, Properties.Resources.SetIotHubDistributedTracing))
+            {
+                IotHubDescription iotHubDescription;
+                if (ParameterSetName.Equals(InputObjectParameterSet))
+                {
+                    this.ResourceGroupName = this.InputObject.Resourcegroup;
+                    this.IotHubName = this.InputObject.Name;
+                    iotHubDescription = IotHubUtils.ConvertObject<PSIotHub, IotHubDescription>(this.InputObject);
+                }
+                else
+                {
+                    if (ParameterSetName.Equals(ResourceIdParameterSet))
+                    {
+                        this.ResourceGroupName = IotHubUtils.GetResourceGroupName(this.ResourceId);
+                        this.IotHubName = IotHubUtils.GetIotHubName(this.ResourceId);
+                    }
+
+                    iotHubDescription = this.IotHubClient.IotHubResource.Get(this.ResourceGroupName, this.IotHubName);
+                }
+
+                IEnumerable<SharedAccessSignatureAuthorizationRule> authPolicies = this.IotHubClient.IotHubResource.ListKeys(this.ResourceGroupName, this.IotHubName);
+                SharedAccessSignatureAuthorizationRule policy = IotHubUtils.GetPolicy(authPolicies, PSAccessRights.RegistryWrite);
+                PSIotHubConnectionString psIotHubConnectionString = IotHubUtils.ToPSIotHubConnectionString(policy, iotHubDescription.Properties.HostName);
+                RegistryManager registryManager = RegistryManager.CreateFromConnectionString(psIotHubConnectionString.PrimaryConnectionString);
+
+                PSDistributedTracing psDistributedTracing = new PSDistributedTracing();
+                Twin deviceTwin = registryManager.GetTwinAsync(this.DeviceId).GetAwaiter().GetResult();
+
+                IotHubDataPlaneUtils.Validate_Device_Tracing(this.DeviceId, iotHubDescription.Sku.Tier.Value.ToString(), iotHubDescription.Location, deviceTwin.Capabilities.IotEdge);
+                
+                if (deviceTwin.Properties.Desired.Contains(IotHubDataPlaneUtils.TRACING_PROPERTY))
+                {
+                    psDistributedTracing = JsonConvert.DeserializeObject<PSDistributedTracing>(deviceTwin.Properties.Desired[IotHubDataPlaneUtils.TRACING_PROPERTY].ToString());
+                }
+
+                psDistributedTracing.SamplingMode = this.SamplingMode;
+
+                if (this.SamplingMode.Equals(PSDistributedTracingSamplingMode.Enabled))
+                {
+                    if (this.MyInvocation.BoundParameters.ContainsKey("SamplingRate"))
```

Comment:
This can avoid hard code string ;)
```suggestion
                    if (this.IsParameterBound(c => c.SamplingRate))
```

Reply:
Removed hardcoded string